Hubble-Relay: proxy context from originating client

nathanjsweet opened the pull request:
The context that comes with the GRPC request has valuable
metadata information in it that is useful for things like
authentication and authorization for downstream servers.

Signed-off-by: Nate Sweet <nathanjsweet@pm.me>

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

<!-- Description of change -->


```release-note
Hubble-relay now proxies the GRPC context to its servers.
```

glibsm commented on pkg/hubble/relay/observer/server.go:
```diff
@@ -164,7 +164,7 @@ sortedFlowsLoop:
 // the ServerStatus answer of all hubble peers.
 func (s *Server) ServerStatus(ctx context.Context, req *observerpb.ServerStatusRequest) (*observerpb.ServerStatusResponse, error) {
-	ctx, cancel := context.WithCancel(context.Background())
+	ctx, cancel := context.WithCancel(ctx)
```
shadow var here is not great

nathanjsweet committed:
Hubble-Relay: proxy context from originating client
The context that comes with the grpc request has valuable
metadata information in it that is useful for things like
authentication and authorization for downstream servers.

Signed-off-by: Nate Sweet <nathanjsweet@pm.me>
